torch/csrc/jit/ir.cpp

```diff
@@ -250,6 +250,175 @@ std::ostream& operator<<(std::ostream & out, const Graph & g) {
   return out;
 }
 
+std::string getBaseName(std::string str) {
+  return str.substr(0, str.find_first_of('.'));
+}
+
+std::ostream& prettyPrintValue(std::ostream & out, const Value* val) {
+  auto scope = val->node()->scope();
+  auto unique = val->uniqueName();
+  if (val->node()->kind() == prim::Constant) {
+    if (scope->hasConstant(unique)) {
+      out << scope->getConstant(unique);
+    } else {
+      throw std::runtime_error("const not found for : " + unique);
+    }
+  } else {
+    auto name = val->readableName();
+    if (scope->hasNameAlias(unique)) {
+      name = scope->getNameAlias(unique);
+    }
+
+    if (isdigit(name.at(0))) {
+      out << "%";
+    }
+    out << name;
+  }
+return out;
+}
+
+std::ostream& prettyPrintInputs(std::ostream & out, const Node* node) {
+  out << "(";
+  auto delimiter = "";
+  for (auto in_value : node->inputs()) {
+    out << delimiter;
+    prettyPrintValue(out, in_value);
+    delimiter = ", ";
+  }
+  out << ")";
+  return out;
+}
+
+std::ostream& prettyPrintNodeVisitor(std::ostream & out, const Node* root, int level);
+
+std::ostream& prettyPrintBlockVisitor(std::ostream & out, const Block* root, int level) {
+  for (auto node : root->nodes()) {
+    prettyPrintNodeVisitor(out, node, level);
+  }
+
+  prettyPrintNodeVisitor(out, root->return_node(), level);
+
+  return out;
+}
+
+// Handles iterating over nodes/subblocks
+std::ostream& prettyPrintNodeVisitor(std::ostream & out, const Node* node, int level) {
+  if (node->kind() == prim::Constant) {
+    // Add constant to current scope map
+    node->scope()->addConstant(
+      node->outputs()[0]->uniqueName(),
+      node->i(node->attributeNames()[0])
+    );
+    return out;
+  }
+
+  if (node->kind() == prim::Return) {
+    // Handled elsewhere, do nothing
+    return out;
+  }
+
+  indent(out, level);
+  int new_level = level + 1;
+
+  // if there are subblocks on this node, visit them
+  switch (node->kind()) {
+  case prim::Loop: {
+    out << "while ";
```

Comment:
I feel like these huge switch cases make it *extremely* easy to forget the `break` at the end. Maybe put each into a little helper function? Or some other solution to this problem